pcsx2/GS/Renderers/HW/GSTextureCache.cpp

```diff
@@ -1459,7 +1459,10 @@ GSTextureCache::Target* GSTextureCache::LookupTarget(GIFRegTEX0 TEX0, const GSVe
 		{
 			const bool forced_preload = GSRendererHW::GetInstance()->m_force_preload > 0;
 			const GSVector4i newrect = GSVector4i::loadh(size);
-			const u32 rect_end = GSLocalMemory::m_psm[TEX0.PSM].info.bn(newrect.z - 1, newrect.w - 1, TEX0.TBP0, TEX0.TBW);
+			u32 rect_end = GSLocalMemory::GetEndBlockAddress(TEX0.TBP0, TEX0.TBW, TEX0.PSM, newrect);
+			if (rect_end < TEX0.TBP0)
+				rect_end += 0x4000;
```

Comment:
Could you use the `MAX_BLOCKS` constant here? (just makes it easier to read)

Reply:
I could do, yeah, I'll modify that later.